lib/src/components/NxCollapsibleItems/__tests__/NxCollapsibleItems.test.tsx

```diff
@@ -32,262 +29,261 @@ describe('NxCollapsibleItems', function() {
+    expect(childrenEl).toContainElement(childEl);
   });
 
-  it('sets specified role on an nx-collapsible-items__children element', function() {
-    const component = getShallowComponent({ role: 'menu' }),
-        childrenEl = component.find('.nx-collapsible-items__children');
+  it('sets specified role on the element containing the children', function() {
+    const view = quickRender({ role: 'menu', children: <span data-testid="child">foo</span> }),
+        childrenEl = view.getByRole('menu'),
+        childEl = view.getByTestId('child');
 
-    expect(childrenEl).toExist();
-    expect(childrenEl).toHaveProp('role', 'menu');
+    expect(childrenEl).toBeInTheDocument();
+    expect(view.queryByRole('list')).not.toBeInTheDocument();
+    expect(childrenEl).toContainElement(childEl);
   });
 
   describe('actionContent', function() {
-    it('renders specified actionContent and does not trigger onToggleCollapse when actionContent is clicked', () => {
-      const onToggleCollapse = jest.fn();
-      const component = getShallowComponent({ actionContent: <NxIconDropdown isOpen={false} />, onToggleCollapse });
-      const toggle = component.find(NxIconDropdown);
-      expect(toggle).toExist();
-      expect(onToggleCollapse).not.toHaveBeenCalled();
-      toggle.simulate('click');
-      expect(onToggleCollapse).not.toHaveBeenCalled();
-    });
+    it('renders specified actionContent and does not trigger onToggleCollapse when actionContent is clicked',
+        async () => {
+          const user = userEvent.setup(),
+              onToggleCollapse = jest.fn(),
+              view = quickRender({
+                actionContent: <NxIconDropdown title="action" isOpen={false} />,
+                onToggleCollapse
+              });
+
+          // wait for title to initialize
+          await runTimers();
+          const actionToggle = view.getByRole('button', { name: 'action' });
+
+          expect(actionToggle).toBeInTheDocument();
+          expect(onToggleCollapse).not.toHaveBeenCalled();
+
+          await user.click(actionToggle);
+          expect(onToggleCollapse).not.toHaveBeenCalled();
+        }
+    );
   });
 
   describe('NxCollapsibleItems.Child', function() {
     describe('when children is a string', function() {
-      const component = mount(
-        <NxCollapsibleItems.Child id="test-id" className="bar" lang="en">foo</NxCollapsibleItems.Child>),
-          div = component.children().children();
+      const minimalProps = { children: 'foo' },
+          quickRender = rtlRender(NxCollapsibleItems.Child, minimalProps),
+          renderEl = rtlRenderElement(NxCollapsibleItems.Child, minimalProps);
 
-      it('renders a div with the listitem role and the nx-collapsible-items__child class', function() {
-        expect(component.hostNodes()).not.toExist();
-        expect(component.children().hostNodes()).not.toExist();
+      it('renders a top-level element with the listitem role', function() {
```

Comment:
should you also test that the role can be overridden if another role is provided, like in line 182?

Reply:
> should you also test (both here and when the child is an element) that the role can be overridden if another role is provided, like in line 182?

There are actually a lot of nuances around the roles logic here that all deserve to be tested. Good catch. 0c825d96a